proparse/src/main/java/org/prorefactor/treeparser/SymbolScope.java

```diff
@@ -327,9 +334,12 @@ public TableBuffer lookupBuffer(String inName) {
     TableBuffer symbol = bufferMap.get(bufferPart.toLowerCase());
     if (symbol == null
         || (dbPart.length() != 0 && !dbPart.equalsIgnoreCase(symbol.getTable().getDatabase().getName()))) {
-      if (parentScope == null)
-        return null;
-      return parentScope.lookupBuffer(inName);
+      if (parentScope != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1066)